phpBB/phpbb/storage/storage.php

```diff
@@ -133,11 +140,64 @@ public function rename($path_orig, $path_dest)
 	 * @param string	$path_orig	The original filename
 	 * @param string	$path_dest	The target filename
 	 *
-	 * @throws \phpbb\storage\exception\exception	When target exists
-	 * 												When the file cannot be copied
+	 * @throws \phpbb\storage\exception\exception		When target exists
+	 * 													When the file cannot be copied
+	 * @throws \phpbb\storage\exception\not_implemented	When the adapter doesnt implement the method
```

Comment:
actually I don't think we should ever throw a not_implemented exception. These methods are mandatory and can easily be faked (copy = create new file ; rename = create new file + delete old etc...)